engines/engine.cpp

```diff
@@ -641,15 +661,73 @@ bool Engine::canLoadGameStateCurrently() {
 }
 
 Common::Error Engine::saveGameState(int slot, const Common::String &desc) {
-	// Do nothing by default
-	return Common::kNoError;
+	return saveGameState(slot, desc, false);
+}
+
+Common::Error Engine::saveGameState(int slot, const Common::String &desc, bool isAutosave) {
+	Common::OutSaveFile *saveFile = _saveFileMan->openForSaving(getSaveStateName(slot));
+
+	if (!saveFile)
+		return Common::kWritingFailed;
+
+	Common::Error result = saveGameStream(saveFile, isAutosave);
+	if (result.getCode() == Common::kNoError) {
+		MetaEngine::appendExtendedSave(saveFile, getTotalPlayTime() / 1000, desc);
+
+		saveFile->finalize();
+	}
+
+	delete saveFile;
+	return result;
+}
+
+Common::Error Engine::saveGameStream(Common::WriteStream *stream, bool isAutosave) {
+	// Default to returning an error when not implemented
+	return Common::kWritingFailed;
 }
 
 bool Engine::canSaveGameStateCurrently() {
 	// Do not allow saving by default
 	return false;
 }
 
+bool Engine::loadGameDialog() {
+	if (!canLoadGameStateCurrently())
```

Comment:
I'm not sure where we are going with this check. It seems to me it can create situations where the player clicks on the load button and nothing happens because loading is not allowed currently. So maybe this should be an assertion to force the caller handle the case or maybe this should display a GUI error message telling the player loading is no possible at the moment.

Reply:
Or an OSD message?

Reply:
Good point both of you; that'd make more sense to users that simply ignoring the call.